src/eventbridge/bus.ts

```diff
@@ -0,0 +1,114 @@
+import { aws_events } from "aws-cdk-lib";
+import { Construct } from "constructs";
+import { EventBusRule, EventPredicateFunction } from "./rule";
+import { EventBusRuleInput } from "./types";
+
+/**
+ * A Functionless wrapper for a AWS CDK {@link aws_events.EventBus}.
+ *
+ * Filtering events and sending them to Lambda.
+ *
+ * ```ts
+ * interface Payload {
+ *    value: string;
+ * }
+ *
+ * // An event with the payload
+ * type myEvent = EventBusRuleInput<Payload>;
```

Comment:
probably better to use an `interface` here

https://blog.logrocket.com/types-vs-interfaces-in-typescript/